src/Controls/src/Core/BindablePropertyConverter.cs

```diff
@@ -23,6 +24,11 @@ public override bool CanConvertTo(ITypeDescriptorContext context, Type destinati
 
 		object IExtendedTypeConverter.ConvertFromInvariantString(string value, IServiceProvider serviceProvider)
 		{
+			if (!RuntimeFeature.IsXamlRuntimeParsingSupported)
+			{
+				throw new InvalidOperationException(RuntimeFeature.XamlRuntimeParsingNotSupportedErrorMessage);
+			}
```

Comment:
It looks like the converter has a XamlC (build-type) replacement:

https://github.com/dotnet/maui/blob/d87c18152fab1d682e505c915106b632934099cf/src/Controls/src/Core/BindablePropertyConverter.cs#L15-L16

So it seems we should be fine to throw here.